### DISCUSSION TITLE
Add block hash in the response of "matches" API?

### DISCUSSION THREAD

**Discussion by cffls** [CardanoSolutions/kupo#5] [Q&A] [answered]
Currently, responses returned by `matches` doesn't not have block hash. Since the response already contains the slot in which the transaction lives, my naive guess is that it wouldn't be much work to add block hash in the response?

The reason that I would want to see block hash in the returned value is that so I can query the entire block using both block hash and slot number in Ogmios (findIntersect API). My main use case is to get the actual datum sitting at a specific address by using only Kupo and Ogimios, without running additional sync service or database, such as cardano-db-sync (I try to avoid adding new services as much as possible). Do you think this usage pattern is reasonable?

An alternative solution to such use case is to add datum directly in the response of `matches` API if they present in the witness of the transaction, but this might be a bigger change than simply adding the block hash.

**Comment by KtorZ** [comment, marked as answer]
Both changes are possible and make a lot of sense. The former however (adding block hashes) seem a bit more in the spirit of Kupo and the lightweight goal.

Using then Ogmios to get data out of the chain-sync protocol is actually a great idea! I originally thought of Kupo used mainly in conjunction with the state query protocol but, it can work indeed very well with the chain-sync protocol provided that the full points description is available.

Thanks for the feedback!

**Reply by cffls** [reply]
Thanks for the response and confirmation. Kupo + Ogmios will very likely be a super fast and lightweight solution, because we can now save tremendously amount of sync time and hardware resources incurred by db-sync otherwise. Really looking forwards to seeing these features in Kupo! :)

**Comment by KtorZ** [comment]
This is now available on `master` and will be released with 1.0.0 though I still have few items to go through before (see https://github.com/CardanoSolutions/kupo/projects/1)

**Reply by cffls** [reply]
Awesome!! I can't wait to use the feature in the next release. Thank you so much for adding this feature in such a short time.

**Reply by KtorZ** [reply]
(don't forget to mark the discussion as resolved once you've been able to try it out)
